### PR TITLE
Cds and UI updates

### PR DESCRIPTION
**Pull request by fanimator12**
- Search bar permanently visible
- header permanently visible
- Drawer fixed
- fullscreen feature for videos fixed
- 'Events' added to the top banner
- 'Indico' changed to 'Events' in the footer
- imprints added in the backend
- CDS records sorted by date (newest come first)
- about us page changed
- duration added to videocards
- new migration

### CODE REVIEW

File: backend/cds/migrations/0002_alter_lecture_lecture_id.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("cds", "0001_initial"),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name="lecture",
+            name="lecture_id",
```
Comment (drjova) [inline backend/cds/migrations/0002_alter_lecture_lecture_id.py:15]: why do you need we need `lecture_id`?

File: backend/cds/views.py [hunk 2]
```diff
@@ -49,4 +49,4 @@ class LectureDocumentView(DocumentViewSet):
         "lecture_id": None,
     }
 
-    ordering = ("lecture_id",)
+    ordering = ("-date",)
```
Comment (drjova) [inline backend/cds/views.py:52]: you don't have to specify it in both places (line 18)

File: ui/src/components/SUGGESTION_BOX.tsx [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
 
-const handleKeyPress = (ev: React.KeyboardEvent<HTMLTextAreaElement>) => {
-  console.log("handleKeyPress", ev);
-};
-
 function SUGGESTION_BOX() {
+  const handleKeyPress = (ev: React.KeyboardEvent<HTMLTextAreaElement>) => {
+    console.log("handleKeyPress", ev);
+  };
+
```
Comment (drjova) [inline ui/src/components/SUGGESTION_BOX.tsx:12]: actually we don't need this, it does nothing

File: ui/src/components/SUGGESTION_BOX.tsx [hunk 4]
```diff
@@ -1,16 +1,17 @@
...
   return (
-    <div className="suggestion-box">
+    <div className={"suggestion-box"}>
```
Comment (drjova) [inline ui/src/components/SUGGESTION_BOX.tsx:14]:
```suggestion
    <div className="suggestion-box">
```

File: ui/src/routes/lectures.tsx [hunk 5]
```diff
@@ -15,6 +15,8 @@ function Lectures() {
 
   const [lectures, setLectures] = React.useState<any>();
 
+  window.scrollTo(0, 0);
```
Comment (drjova) [inline ui/src/routes/lectures.tsx:18]: why do we need this?

File: ui/src/routes/results.tsx [hunk 6]
```diff
@@ -50,6 +50,8 @@ function Results() {
     setPageSize(size);
   };
 
+  window.scrollTo(0, 0);
+
```
Comment (drjova) [inline ui/src/routes/results.tsx:54]: why do we need this?

File: ui/src/routes/results.tsx [hunk 7]
```diff
@@ -96,6 +98,9 @@ function Results() {
                                   alt="thumbnail"
                                   src={lecture.thumbnail_picture}
                                 />
+                                {/* <div className="duration">
+                                  <Title level={5}>{lecture.imprint.substring(12, 20)}</Title>
+                                </div> */}
```
Comment (drjova) [inline ui/src/routes/results.tsx:103]: you can completely remove this comment

File: backend/cds/migrations/0002_alter_lecture_lecture_id.py (hunk 1 above)
Comment (fanimator12) [reply backend/cds/migrations/0002_alter_lecture_lecture_id.py:15]: we don't, I didn't notice it was there

File: ui/src/components/SUGGESTION_BOX.tsx (hunk 3 above)
Comment (fanimator12) [reply ui/src/components/SUGGESTION_BOX.tsx:12]: The suggestion box does not send yet anything anywhere, but we need it later. We discuss this with Maria.

File: ui/src/routes/lectures.tsx (hunk 5 above)
Comment (fanimator12) [reply ui/src/routes/lectures.tsx:18]: Because it should bring the user to the top of the page.

File: ui/src/routes/results.tsx (hunk 6 above)
Comment (fanimator12) [reply ui/src/routes/results.tsx:54]: Because it should bring the user to the top of the page.

File: backend/cds/migrations/0001_initial.py [hunk 8]
```diff
@@ -22,7 +22,7 @@ class Migration(migrations.Migration):
                         verbose_name="ID",
                     ),
                 ),
-                ("lecture_id", models.IntegerField()),
+                ("lecture_id", models.IntegerField(unique=True)),
```
Comment (drjova) [inline backend/cds/migrations/0001_initial.py:25]: why you changed that? 🤔

File: ui/src/App.tsx [hunk 9]
```diff
@@ -55,19 +57,15 @@ function App() {
...
+                    <div key={photo.src} className="container-fluid">
+                      <div className="content">
+                        <img alt={photo.alt} src={photo.src} />
                       </div>
-                    );
-                  })
-                )}
+                    </div>
+                  );
+                })}
```
Comment (drjova) [inline ui/src/App.tsx:68]: why did you remove the the loading?

File: ui/src/App.tsx [hunk 10]
```diff
@@ -128,6 +126,12 @@ function App() {
                               <PlayCircleOutlined />
                             </div>
 
+                            <div className="duration">
+                              <Title level={5}>
+                                {lecture.imprint.substring(12, 20)}
```
Comment (drjova) [inline ui/src/App.tsx:131]: why subsstring? what is the original value?

File: ui/src/components/MENU.tsx [hunk 11]
```diff
@@ -0,0 +1,50 @@
+import React, { useState } from "react";
+import { Link as LinkRouter } from "react-router-dom";
+import { Typography, Menu } from "antd";
+import "./AT_HEADER.css";
+import "antd/dist/antd.css";
+
+const { Title } = Typography;
+
+function MENU() {
+  const [current, setCurrent] = React.useState("icon");
```
Comment (drjova) [inline ui/src/components/MENU.tsx:10]:
```suggestion
  const [current, setCurrent] = useState("icon");
```

File: ui/src/components/MENU.tsx [hunk 12]
```diff
@@ -0,0 +1,50 @@
+import React, { useState } from "react";
+import { Link as LinkRouter } from "react-router-dom";
+import { Typography, Menu } from "antd";
+import "./AT_HEADER.css";
+import "antd/dist/antd.css";
```
Comment (drjova) [inline ui/src/components/MENU.tsx:5]: you don't need to import `antd.css` in all files :) could you please just leave it in the `index.js`?

File: ui/src/routes/about_us.tsx [hunk 13]
```diff
@@ -381,11 +394,17 @@ function AboutUs() {
                 );
               })}
             </Row>
-            <div className="atc-title">
+            .{" "}
```
Comment (drjova) [inline ui/src/routes/about_us.tsx:397]: do we need the `.`?

File: backend/cds/migrations/0001_initial.py (hunk 8 above)
Comment (fanimator12) [reply backend/cds/migrations/0001_initial.py:25]: I accidentally removed the wrong migration earlier and undone the initial one. Did not notice that it put extra stuff there that is not needed. I am going to remove this. Thanks for the notice.

File: ui/src/App.tsx (hunk 9 above)
Comment (fanimator12) [reply ui/src/App.tsx:68]: I was told to remove the Spin. Should I put it back?

File: ui/src/App.tsx (hunk 10 above)
Comment (fanimator12) [reply ui/src/App.tsx:131]: The original value is for "imprint" is "2021-10-27 - 1:06:01" for example. I only need the duration.

File: ui/src/routes/about_us.tsx (hunk 13 above)
Comment (fanimator12) [reply ui/src/routes/about_us.tsx:397]: nope, that shouldn't be here - I remove it

File: ui/src/routes/lectures.tsx (hunk 5 above)
Comment (fanimator12) [reply ui/src/routes/lectures.tsx:17]: When it opens a new page or triggers a new route. For example, if the user clicks on a video, the new page with the video should start from the top of the page, not in the middle or at the last position where the user left.

File: ui/src/App.tsx (hunk 9 above)
Comment (drjova) [reply ui/src/App.tsx:68]: 👍  I wasn't aware of that

File: backend/cds/migrations/0001_initial.py [hunk 14]
```diff
@@ -1,4 +1,4 @@
-# Generated by Django 3.2.8 on 2022-02-09 15:40
+# Generated by Django 3.2.8 on 2022-05-19 12:07
```
[the remark this answers is not in the document]
Comment (fanimator12) [reply backend/cds/migrations/0001_initial.py]: sure

File: ui/src/App.tsx (hunk 10 above)
Comment (fanimator12) [reply ui/src/App.tsx:131]: Genial. Thanks :D
